packages/core/src/resource/resource.ts

```diff
@@ -173,11 +180,15 @@ export class ResourceImpl<T, R> extends BaseWritableResource<T> implements Resou
   private destroyed = false;
   private unregisterOnDestroy: () => void;
 
+  override readonly status: Signal<ResourceStatus>;
+  override readonly error: Signal<Error | undefined>;
```

Comment:
Since this is only a type override we can prevent ts from creating additional props. 

```suggestion
  declare readonly status: Signal<ResourceStatus>;
  declare readonly error: Signal<Error | undefined>;
```

Reply:
Not sure I follow that. These are abstract properties that I need to define in the constructor. The `override` is required since our linter doesn't allow implicit overrides. The props should be created in the `ResourceImpl` class.